source/common/ssl/ssl_socket.cc

```diff
@@ -226,6 +227,35 @@ std::string SslSocket::sha256PeerCertificateDigest() {
   return Hex::encode(computed_hash);
 }
 
+std::string SslSocket::peerCertificate() const {
+  bssl::UniquePtr<X509> cert(SSL_get_peer_certificate(ssl_.get()));
+  if (!cert) {
+    return "";
+  }
+
+  int length;
+  unsigned char* str = nullptr;
+  length = ASN1_item_i2d(reinterpret_cast<ASN1_VALUE*>(cert.get()), &str, ASN1_ITEM_rptr(X509));
+  if (!str) {
```

Comment:
I will defer to @ggreenway or @PiotrSikora on this boring ssl code here, but in general we prefer to use `RELEASE_ASSERT` and crash for calls that should never reasonably fail (please see the error handling guide in STYLE.md). I think everything other than the initial call to get the peer certificate is in that category.